blockstore/apps/bundles/tests/test_store.py

```diff
@@ -93,3 +89,57 @@ def test_multiple_snapshots(self, snapshot_created_mock):
             snapshot_1.files['test.txt'].hash_digest,  # pylint: disable=unsubscriptable-object
             snapshot_2.files['renamed.txt'].hash_digest,  # pylint: disable=unsubscriptable-object
         )
+
+
+@mock.patch('blockstore.apps.bundles.store.snapshot_created.send')
+class TestDrafts(TestCase):
+    """
+
+    """
+    def test_basic_creation(self, snapshot_created_mock):
+        bundle_uuid = uuid.UUID('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')
+        snapshot_store = SnapshotStore()
+        draft_store = DraftStore(snapshot_store)
+        draft_name = 'studio_draft'
+        file_mapping = {
+            'test.html': HTML_FILE,
+            'test.txt': TEXT_FILE,
+        }
+        snapshot = snapshot_store.create(bundle_uuid, file_mapping)
+        draft = draft_store.create(bundle_uuid, draft_name, snapshot)
+
+        # Make sure the persisted Draft is retrieved properly.
+        self.assertEqual(draft, draft_store.get(bundle_uuid, draft_name))
+
+        # The Draft was just created and has no changes, so the files it has
+        # should be identical to what's already in the Snapshot it's based on.
+        self.assertEqual(snapshot.files, draft.files)
+
+        # Any file that a Draft doesn't overwrite should return the same URL as
+        # what's in the Snapshot.
+        for file_path in draft.files:
+            self.assertEqual(
+                draft_store.url(draft, file_path),
+                snapshot_store.url(snapshot, file_path)
+            )
+
+        second_draft = draft_store.update_files(
+            bundle_uuid=bundle_uuid,
+            name=draft_name,
+            updated_files={
+                'test.txt': ContentFile(b"This is draft text content"),
+                'test.rst': ContentFile(b"This is draft RST content")
+            }
+        )
+
+        # Make sure the second draft persisted properly.
+        self.assertEqual(second_draft, draft_store.get(bundle_uuid, draft_name))
+
+        # Create a new Snapshot from the Draft
+        with draft_store.file_mapping(second_draft) as draft_files:
+            second_snapshot = snapshot_store.create(draft.bundle_uuid, draft_files)
```

Comment:
Is there some way this could be `second_snapshot = draft_store.commit(draft_name)` or similar, pushing this logic into `DraftStore`?

Reply:
Yeah, you're right. That's a better interface for it, since `DraftStore` already has a reference to the backing `SnapshotStore`.